server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -603,20 +607,16 @@ private static NamedAnalyzer produceAnalyzer(String name,
         return analyzer;
     }
 
-    private void processNormalizerFactory(
+    private static void processNormalizerFactory(
             String name,
             AnalyzerProvider<?> normalizerFactory,
             Map<String, NamedAnalyzer> normalizers,
-            String tokenizerName,
-            TokenizerFactory tokenizerFactory,
+            Supplier<TokenizerFactory> tokenizerSupplier,
             Map<String, TokenFilterFactory> tokenFilters,
             Map<String, CharFilterFactory> charFilters) {
-        if (tokenizerFactory == null) {
-            throw new IllegalStateException("keyword tokenizer factory is null, normalizers require analysis-common module");
```

Comment:
Should we keep this check and move it inside the function further down that calls tokenizerSupplier.get()? I'm not really sure its useful, but it looks like it might provide useful hints about a misconfigured cluster (thats the only way I can see analysis-common missing). In case you re-add this, this should get a test too.

Reply:
If we're missing analysis-common then I don't think anything is going to work?  The error message here is out of date anyway, as we build normalizers with both keyword and whitespace tokenizers, but I'll change the caller to pass `KeywordTokenizer::new` rather than calling into the tokenizer map to ensure that this would never be a problem.